packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -558,7 +612,9 @@ class ScaffoldState extends State<Scaffold> {
 
   @override
   Widget build(BuildContext context) {
-    final EdgeInsets padding = MediaQuery.of(context).padding;
+    EdgeInsets padding = MediaQuery.of(context).padding;
+    if (!config.resizeToAvoidWindowPadding)
+      padding = new EdgeInsets.only(top: padding.top);
```

Comment:
Maybe we should rename it to resizeToAvoidBottomPadding?
Although consider if Android were to use bottom padding to keep us out of the soft buttons. Presumably the padding would be non-zero without a keyboard, and merely bigger with a keyboard. So really it's about not responding to padding changes? I don't really know what the right solution is here but I feel like what we're doing right now isn't fully coherent.